### ISSUE TITLE
(pc-linux) psp should not use signals (1hz timer)

### ISSUE THREAD

**Issue by skliper** [nasa/PSP#88]
The PSP 1hz timer uses SIGALARM and setitimer. There is a proposal ticket:76 to use timer_create() instead. This still has the problem  of using asynchronous signal delivery. Asynchronous signal delivery does not belong in realtime applications. We should remove this and instead us timerfd_create() and create a thread to wait on the timer event. This tracks overruns as well as removes the danger of calling thread-unsafe functions in signal handlers (which pc-linux does.)

In order to avoid the race condition specified in ticket:23 the PSP API should be changed for the user to register a callback rather than depeding on an external dependency from TIME. This would allow PSP to start the timer relative to "bootup" and it would allow the user to avoid the race condition by registering itself with the timer. If we made it a blocking call the user could call from their own thread, it also would let the user control the priority of the thread receiving the timer event.

I believe the point of the 1hz timer is for platforms where there is a hardware timer that can be used with better granulatiry than OS timers. If that is no longer true, then we should just get rid of this and use an OS timer.

**Comment by skliper** [comment]
_Imported from trac issue 84.  Created by sseeger on 2018-06-12T11:27:33, last modified: 2019-06-26T14:26:35_

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-05-06 13:00:04:_

Marking for CCB review, related to #79 #26   Or even better, architecture review.

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-05-06 15:29:35:_

Subscribing.

Note that the solution proposed in #26 also addresses this because when combined with an "ng" osal (i.e. something that implements the "timebase" feature as proposed), the timebase also provides a proper non-signal context for callbacks.

If we agree on the OSAL-NG and #26  then I would say this is solved too.

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-06-13 14:35:32:_

Marked for CCB review - suggest closure (duplicate) to #26

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-06-26 12:33:54:_

CCB 6/26/19 - Agreed to close this ticket based on the old implementation.  New implementation (from #26  will be reviewed to see if there is any further concerns.

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-06-26 14:26:35:_

As noted above this ticket is OBE, closing as "wontfix" since it wasn't addressed directly and the new implementation in #26 still uses signals but in a different way.
